EngineLayer/Psm.cs

```diff
@@ -401,14 +401,14 @@ internal void Add(Psm psmParent)
         private Tuple<string, double?> Resolve(IEnumerable<double> enumerable)
         {
             double doubleTol = 1e-6;
-            var list = enumerable.ToList();
+            var list = enumerable.Distinct().ToList();
```

Comment:
IMO it is more confusing to have the same protein repeated several times per peptide, and it makes it look like we don't know how to parse a database. it's a matter of readability/usability. maximum information vs least confusion.